pkg/ebpf/uprobes/attacher.go

```diff
@@ -0,0 +1,721 @@
+	"github.com/DataDog/datadog-agent/pkg/process/monitor"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+// ExcludeMode defines the different optiont to exclude processes from attachment
+type ExcludeMode uint8
+
+const (
+	// ExcludeSelf excludes the agent's own PID
+	ExcludeSelf ExcludeMode = 1 << iota
+	// ExcludeInternal excludes internal DataDog processes
+	ExcludeInternal
+	// ExcludeBuildkit excludes buildkitd processes
+	ExcludeBuildkit
+	// ExcludeContainerdTmp excludes containerd tmp mounts
+	ExcludeContainerdTmp
+)
+
+const procFSUpdateTimeout = 10 * time.Millisecond
+
+var (
+	// ErrSelfExcluded is returned when the PID is the same as the agent's PID.
+	ErrSelfExcluded = errors.New("self-excluded")
+	// ErrInternalDDogProcessRejected is returned when the PID is an internal datadog process.
+	ErrInternalDDogProcessRejected = errors.New("internal datadog process rejected")
+	// ErrNoMatchingRule is returned when no rule matches the shared library path.
+	ErrNoMatchingRule = errors.New("no matching rule")
+	// regex that defines internal DataDog processes
+	internalProcessRegex = regexp.MustCompile("datadog-agent/.*/((process|security|trace)-agent|system-probe|agent)")
+)
+
+// AttachTarget defines the target to which we should attach the probes, libraries or executables
+type AttachTarget uint8
+
+const (
+	// AttachToExecutable attaches to the main executable
+	AttachToExecutable AttachTarget = 1 << iota
+	// AttachToSharedLibraries attaches to shared libraries
+	AttachToSharedLibraries
+)
+
+// AttachRule defines how to attach a certain set of probes. Uprobes can be attached
+// to shared libraries or executables, this structure tells the attacher which ones to
+// select and to which targets to do it.
+type AttachRule struct {
+	// LibraryNameRegex defines which libraries should be matched by this rule
+	LibraryNameRegex *regexp.Regexp
+	// Targets defines the targets to which we should attach the probes, shared libraries and/or executables
+	Targets AttachTarget
+	// ProbesSelectors defines which probes should be attached and how should we validate
+	// the attachment (e.g., whether we need all probes active or just one of them, or in a best-effort basis)
+	ProbesSelector []manager.ProbesSelector
+}
+
+func (r *AttachRule) canTarget(target AttachTarget) bool {
+	return r.Targets&target != 0
+}
+
+func (r *AttachRule) matchesLibrary(path string) bool {
+	return r.canTarget(AttachToSharedLibraries) && r.LibraryNameRegex != nil && r.LibraryNameRegex.MatchString(path)
+}
+
+func (r *AttachRule) matchesExecutable(_ string) bool {
+	return r.canTarget(AttachToExecutable)
+}
+
+// AttacherConfig defines the configuration for the attacher
+type AttacherConfig struct {
+	// Rules defines a series of rules that tell the attacher how to attach the probes
+	Rules []*AttachRule
+
+	// ScanTerminatedProcessesInterval defines the interval at which we scan for terminated processes. Set
+	// to zero to disable
```

Comment:
I think that's a pitfall, as zero is the default value
we should ensure engineers explicitly disable the termination process, so I'd choose a different value / bool

Reply:
also, the documentation does no match the implementation in `SetDefaults`
I think the implementation is perfect, so we just need to change the documentation